### DISCUSSION TITLE
How to get information about particular variable type and read_only flag?

### DISCUSSION THREAD

**Discussion by mateusz-glowinski** [dmroeder/pylogix#218] [Q&A]
I'm using `Read` method to get variable value.
Such a response, however, seems to have no information about the type of variable or information about whether I can write something to it or not (if variable is flagged as `read only`).

Is there a way to get such an information related to particular variable?

**Comment by dmroeder** [comment]
Try GetTagList(), find your TagName in the returned results. Look at AccessRights

https://github.com/dmroeder/pylogix/blob/master/pylogix/lgx_tag.py#L32

**Reply by mateusz-glowinski** [reply]
Thank you @dmroeder for a tip!

Nevertheless, I've one more question.
For Micro820 I always get `None` as `AccessRight`, even for `_IO_EM_DO_XX` inputs.
Do you maybe know, which value should i expect here when a variable is marked as `read only` in PLC?

**Comment by dmroeder** [comment]
For some unknown reason, those status tags seem inaccessible externally in the Micro800 platform.  Same with local variables. There are a number of things that they did right with that platform, better than the high end controllers, but they limited other things for some reason.

AccessRights none mean no external access, so it won't return a value. There is an AccessRights read only and read/write. If the tag is read only, you should be able to read it, just not write. The status bits on micro800 might be different though

**Reply by mateusz-glowinski** [reply]
Thank for explanation.


> There is an AccessRights read only and read/write.

So, for `ControlLogix` family, I will get `read only` string as `AccessRight` attribute value when PLC variable is read only?
Sorry for primitive question, but I don't have `ControlLogix` to check it on my own.

**Reply by mateusz-glowinski** [reply]
bump @dmroeder :)

**Comment by dmroeder** [comment]
Hmm, it appears that AccessRight is only being looked at for UDT definitions, not instances of tags.  I need to do some further digging...

**Reply by mateusz-glowinski** [reply]
ok, thanks for checking :)

**Reply by TheFern2** [reply]
Assuming you don't have access to the plc program to check read/write access. In clx you can also parse the l5x program and check tags attributes. Anyway, one easy way to do a check is to attempt to write to the tags, then do a read and compare. If the value didn't change we can assume is read only. From there you can build your internal list of attributes for the tags.

**Reply by TheFern2** [reply]
What happens if we try writing to a read only, do you still get Success on the status?

**Reply by dmroeder** [reply]
If you try to write to a read only tag the controller will return "Privilege Violation" as the status

And while we're at it, writing while access set to None returns "Path segment error".  Which is the long way around for saying the tag doesn't exist

**Reply by TheFern2** [reply]
Ah ok, so one could technically use those two statuses to assume tag isn't available for write. This is just an idea to use in the meantime, instead of waiting for a new pylogix feature for tag attributes.

**Reply by dmroeder** [reply]
Yeah.  If ret.Status is not "Success"....

I'm not sure what I'll do with this one.  It appears it will be quite a bit of work, after I figure out the solution would be.  There are other things that are more important to me at the moment.

**Reply by mateusz-glowinski** [reply]
Ok, so the summary is that I can't get the `right access` of a tag like i do with `tag type` at this moment ? Even for ControlLogix family?

**Reply by dmroeder** [reply]
You are correct. Sorry I misspoke earlier about this.

**Reply by mateusz-glowinski** [reply]
Ok, thanks for clarification.
Will you create an issue related to this topic or this is minor issue and no time to handle that?

**Reply by dmroeder** [reply]
Currently, I'd say the latter.  I have a few things in the works will keep me busy enough.

**Reply by mateusz-glowinski** [reply]
ok :)
